src/core/fixkey_hash_table.c

```diff
@@ -0,0 +1,302 @@
+#include "moar.h"
+
+#define FIXKEY_LOAD_FACTOR 0.75
+#define FIXKEY_INITIAL_SIZE 8
+#define FIXKEY_INITIAL_KEY_RIGHT_SHIFT (8 * sizeof(MVMuint64) - 3)
+
+MVM_STATIC_INLINE MVMuint32 hash_true_size(MVMFixKeyHashTable *hashtable) {
+    MVMuint32 true_size = hashtable->official_size + hashtable->max_items - 1;
+    if (hashtable->official_size + MVM_HASH_MAX_PROBE_DISTANCE < true_size) {
+        true_size = hashtable->official_size + MVM_HASH_MAX_PROBE_DISTANCE;
+    }
+    return true_size;
+}
+
+/* Frees the entire contents of the hash, leaving you just the hashtable itself,
+   which you allocated (heap, stack, inside another struct, wherever) */
+void MVM_fixkey_hash_demolish(MVMThreadContext *tc, MVMFixKeyHashTable *hashtable) {
```

Comment:
Nitpick, but most (all?) of the other similar functions are `*_destroy`.

Reply:
Oops, I missed this comment. My intent was deliberate - they don't **free** memory. Whereas my understanding of `destroy` was that that meant "and frees the memory".

Reply:
They do tend to, indeed (though there may be an exception somewhere). That's reasonable, anyway. Though I note `uv_mutex_destroy` does work like you described, so we're not in a state of complete `destroy` consistency in the codebase (even if libuv's naming isn't really our deciding :)).